cfme/tests/networks/nuage/test_inventory.py

```diff
@@ -82,3 +82,101 @@ def test_network_router_details(setup_provider_modscope, provider, with_nuage_sa
+    """
+    sandbox = with_nuage_sandbox_modscope
+    tenant_name = sandbox['enterprise'].name
+    subnet_name = sandbox['subnet'].name
+    router_name = sandbox['domain'].name
+    vport_name = sandbox['cont_vport'].name
+    tenant = provider.collections.cloud_tenants.instantiate(name=tenant_name, provider=provider)
+    router = tenant.collections.routers.instantiate(name=router_name)
+    subnet = router.collections.subnets.instantiate(name=subnet_name)
+    network_port = subnet.collections.network_ports.instantiate(name=vport_name)
+
+    network_port.validate_stats({
+        ('properties', 'Name'): vport_name,
+        ('properties', 'Type'): 'ManageIQ/Providers/Nuage/Network Manager/Network Port/Container',
+        ('relationships', 'Cloud tenant'): tenant_name,
+        ('relationships', 'Cloud subnets'): '1',
+    })
+
+
+def test_network_port_l2_vm(setup_provider_modscope, provider, with_nuage_sandbox_modscope):
+    """
+    Ensure vm network port details displays expected info.
+
+    L2 subnets act as standalone and are thus not connected to any router.
+    We navigate to network ports as
+      Tenant > Subnet > Network Port
+    """
+    sandbox = with_nuage_sandbox_modscope
+    tenant_name = sandbox['enterprise'].name
+    subnet_name = sandbox['l2_domain'].name
+    vport_name = sandbox['l2_vm_vport'].name
+    tenant = provider.collections.cloud_tenants.instantiate(name=tenant_name, provider=provider)
+    subnet = tenant.collections.subnets.instantiate(name=subnet_name)
+    network_port = subnet.collections.network_ports.instantiate(name=vport_name)
+
+    network_port.validate_stats({
+        ('properties', 'Name'): vport_name,
+        ('properties', 'Type'): 'ManageIQ/Providers/Nuage/Network Manager/Network Port/Vm',
+        ('relationships', 'Cloud tenant'): tenant_name,
+        ('relationships', 'Cloud subnets'): '1',
+    })
+
+
+def test_network_port_l2_container(setup_provider_modscope, provider, with_nuage_sandbox_modscope):
+    """
+    Ensure container network port details displays expected info.
+
+    L2 subnets act as standalone and are thus not connected to any router.
+    We navigate to network ports as
+      Tenant > Subnet > Network Port
+    """
+    sandbox = with_nuage_sandbox_modscope
+    tenant_name = sandbox['enterprise'].name
+    subnet_name = sandbox['l2_domain'].name
+    vport_name = sandbox['l2_cont_vport'].name
+    tenant = provider.collections.cloud_tenants.instantiate(name=tenant_name, provider=provider)
+    subnet = tenant.collections.subnets.instantiate(name=subnet_name)
+    network_port = subnet.collections.network_ports.instantiate(name=vport_name)
+
+    network_port.validate_stats({
+        ('properties', 'Name'): vport_name,
+        ('properties', 'Type'): 'ManageIQ/Providers/Nuage/Network Manager/Network Port/Container',
+        ('relationships', 'Cloud tenant'): tenant_name,
+        ('relationships', 'Cloud subnets'): '1',
+    })
```

Comment:
@aljazkosir I'm thinking we could extract common into a function as @digitronik suggested:

```python
def test_network_port_l2_container(...)
  assert_network_port(...)

def test_network_port_l2_vm(...)
  assert_network_port(...)

def assert_network_port(subnet_name, vport_name, expected_stats):
  """Helper function"""
  ...
```

Think that would make sense?

Reply:
@aljazkosir 

LGTM; please can you try some parametrize like
```
@pytest.mark.parametrize("port", ["vm_vport", "cont_vport"], ids=["vm", "container"])
@pytest.mark.parametrize("domain", ["default", "l2_domain"])
def test_network_port(port, domain):
    # some manuculation related to port and domain as per test
    print(port, domain)  # print conbination
```
```
test.py::test_network_port[default-vm] ('vm_vport', 'default')
PASSED
test.py::test_network_port[default-container] ('cont_vport', 'default')
PASSED
test.py::test_network_port[l2_domain-vm] ('vm_vport', 'l2_domain')
PASSED
test.py::test_network_port[l2_domain-container] ('cont_vport', 'l2_domain')
PASSED
```
I thing good to  parametrize. Please let me know if you face problem.

Reply:
This won't work since "default" and "l2_domain" ports follow different paths, "l2_domain" doesn't go through router while L3 does. I think it is clearer the way it is.

\cc @miha-plesko @mshriver

Reply:
@digitronik I had a discussion with @aljazkosir and we kind of agreed that in tests it's better to be a more explicit rather than overdo DRY. I understand we're trying to reuse as much as possible, but in this case I'd say the way @aljazkosir did it is just the best...